packages/runtime/runtime-definitions/src/dataStoreContext.ts

```diff
@@ -487,6 +520,9 @@ export interface IFluidDataStoreContext
 	 * @param outboundHandle - The handle of the outbound node that is referenced.
 	 */
 	addedGCOutboundReference?(srcHandle: IFluidHandle, outboundHandle: IFluidHandle): void;
+
+	/** Indicates whether the last known GC run found this object to be referenced or not, and other related details */
+	readonly experimentalGCInfo: IExperimentalFluidGCInfo;
```

Comment:
if you really want it experimental, and to have the freedom to modify it outside major releases you should create a new interface as a peer to this interface, IExperimentalDataStoreContext, where all members are optional.

The implemention then implements both the experimental, and the non-experimental interfaces.

User then need to explictly case to the the experimental interface the leverage it.

The reason you can't modify the real interface is that any change to any member is breaking to the whole interface. If you put this on the real interface, it can't be experimental in my opinion.

For an example, see IContainer and IContainerExperimental

Reply:
I think this would also let you remove experimental from all the individual things, only the top most experimental interface is named as such, this will allow moving to non-experimental to be less breaking, as the members don't change.